cluster-provision/gocli/opts/prometheus/prometheus.go

```diff
@@ -0,0 +1,133 @@
+package prometheus
+
+import (
+	"embed"
+
+	k8s "kubevirt.io/kubevirtci/cluster-provision/gocli/pkg/k8s"
+)
+
+//go:embed manifests/*
+var f embed.FS
+
+type PrometheusOpt struct {
+	grafanaEnabled      bool
+	alertmanagerEnabled bool
+
+	client k8s.K8sDynamicClient
+}
+
+func NewPrometheusOpt(c k8s.K8sDynamicClient, grafanaEnabled, alertmanagerEnabled bool) *PrometheusOpt {
+	return &PrometheusOpt{
+		grafanaEnabled:      grafanaEnabled,
+		alertmanagerEnabled: alertmanagerEnabled,
+		client:              c,
+	}
+}
+
+func (o *PrometheusOpt) Exec() error {
+	defaultManifests := []string{
```

Comment:
To avoid boilerplate here, would it make sense to write the manifests to a temp file system and apply them one by one, iterating over the manifests with [WalkDir](https://pkg.go.dev/io/fs@master#WalkDir)?

Reply:
Yeah that would be alot better indeed i will implement this